intelmq/bots/outputs/redis/output.py

```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
+import sys
+import time
+import redis
+import intelmq.lib.utils as utils
+from intelmq.lib.bot import Bot
+
+
+class RedisOutputBot(Bot):
+
+    def init(self):
+        self.host = str(self.parameters.redis_server_ip)
+        self.port = int(self.parameters.redis_server_port)
+        self.db = int(self.parameters.redis_db)
+        self.queue = str(self.parameters.redis_queue)
+        self.password = str(self.parameters.redis_password)
+        self.timeout = int(self.parameters.redis_timeout)
+
+        self.conn = redis.ConnectionPool(host = self.host, port = self.port, db = self.db)
+
+    def process(self):
+        event = self.receive_message()
+        
+        if event is None:
+            self.acknowledge_message()
+            return
+
+        while True:
```

Comment:
What? The bot class is doing error handling already. It is allowed to fail for `process()`. Waiting and processing an event again is also handled by `bot.py` and `pipeline.py`. You can put the `redis.StrictRedis` in a separate function, which is called on errors and in `init()`

Also, you can deliver othe error-handling parameters in your config, which are adapted for this case.